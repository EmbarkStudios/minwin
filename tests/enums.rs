mod utils;
use utils::*;

// Ensures we can emit some but not all of a simple enum's values
test!(
    simple,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    [
        "Windows.Win32.Foundation.GENERIC_READ",
        "Windows.Win32.Foundation.GENERIC_WRITE",
        "Windows.Win32.Foundation.GENERIC_ALL",
        "Windows.Win32.Storage.FileSystem.FILE_TYPE_DISK",
        "Windows.Win32.Storage.FileSystem.GetFileType",
        "Windows.Win32.Storage.FileSystem.ClfsMgmtLogWriteNotification",
        "Windows.Win32.Storage.FileSystem.ReadLogNotification",
    ]
);

// Ensures we can emit enums that use core types instead of a typedef
test!(
    core,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    ["Windows.Win32.Foundation.S_OK"]
);

// Ensures we can emit the enum type alias, even if no actual enum variants
// are requested
test!(
    always_emits_alias,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    ["Windows.Win32.UI.Input.Pointer.GetPointerTouchInfo"]
);

// Ensures that enums that clash with handle types are supported
test!(
    handle_and_enum,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    [
        "Windows.Win32.UI.WindowsAndMessaging.SetWindowPos",
        "Windows.Win32.UI.WindowsAndMessaging.GetMessageW",
        "Windows.Win32.UI.WindowsAndMessaging.HWND_NOTOPMOST",
    ]
);
