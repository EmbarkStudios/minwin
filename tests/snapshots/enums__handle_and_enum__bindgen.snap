---
source: tests/enums.rs
expression: bo.bindings
---
// Bindings generated by `windows-bindgen` 0.49.0

#![allow(
    non_snake_case,
    non_upper_case_globals,
    non_camel_case_types,
    dead_code,
    clippy::all
)]
::windows_targets::link!("user32.dll" "system" fn GetMessageW(lpmsg : *mut MSG, hwnd : HWND, wmsgfiltermin : u32, wmsgfiltermax : u32) -> BOOL);
::windows_targets::link!("user32.dll" "system" fn SetWindowPos(hwnd : HWND, hwndinsertafter : HWND, x : i32, y : i32, cx : i32, cy : i32, uflags : SET_WINDOW_POS_FLAGS) -> BOOL);
pub type BOOL = i32;
pub type HWND = isize;
pub const HWND_NOTOPMOST: HWND = -2i32 as _;
pub type LPARAM = isize;
#[repr(C)]
pub struct MSG {
    pub hwnd: HWND,
    pub message: u32,
    pub wParam: WPARAM,
    pub lParam: LPARAM,
    pub time: u32,
    pub pt: POINT,
}
impl ::core::marker::Copy for MSG {}
impl ::core::clone::Clone for MSG {
    fn clone(&self) -> Self {
        *self
    }
}
#[repr(C)]
pub struct POINT {
    pub x: i32,
    pub y: i32,
}
impl ::core::marker::Copy for POINT {}
impl ::core::clone::Clone for POINT {
    fn clone(&self) -> Self {
        *self
    }
}
pub type SET_WINDOW_POS_FLAGS = u32;
pub type WPARAM = usize;

